### ISSUE TITLE
migrate discord.ocaml.org to a new cluster

### ISSUE THREAD

**Issue by avsm** [ocaml/v2.ocaml.org#1175]
*See https://github.com/ocaml/ocaml.org/issues/1169 for the overall tracking issue.*

There is a chat bridge (`matterbridge`) which operates as a bot for the OCaml Discord community. This is currently running on a Rackspace host and needs to be migrated away. This is a good chance to see if we can configure it as as a Dockerfile instead of directly on the VM.

I set this up originally in April 2018 (see [infrastructure thread](http://lists.ocaml.org/pipermail/infrastructure/2018-April/000662.html)) in response to a request from @bluddy.

@bluddy: would you be able to comment on whether you can convert this to a Dockerfile easily? Feel free to email me privately on anil@recoil.org if it involves any sensitive infrastructure details.

**Comment by bluddy** [comment]
Off the top of my head, this should be doable. The only thing that needs to
be kept secret is the discord webhook uri. Currently this resides in the VM
itself and in the admin panel on discord. Is there a method for keeping
things secret with a dockerfile?

**Comment by avsm** [comment]
Yes, you can use Docker secrets for this (https://docs.docker.com/engine/swarm/secrets/). It's essentially just a file on the container filesystem, so if you could create a Dockerfile that assumes the webhook uri is in a particular place, I can take care of hooking it into the cluster deployment.

**Comment by avsm** [comment]
@bluddy, do you have a sense of when you can get me this Dockerfile?  I need to plan the migration to the new cluster.

**Comment by bluddy** [comment]
I've started it but haven't yet had a chance to test it out.

**Comment by avsm** [comment]
Checking in on this. The Discord server is one of the few things keeping me paying for the Rackspace Cloud account...

**Comment by avsm** [comment]
@bluddy - I'm going to have to destroy this virtual machine in a month or so.  If you don't get back to me, I'll recreate a new Discord bridge, but the old secret will be gone.

**Comment by bluddy** [comment]
@avsm I replied to your private email.
